assignment-4.md

```diff
@@ -1,19 +1,24 @@
 # Assignment 4: Application State with Redux
 
 Up to now you have structured your application in a way that holds the state of
-the individual components locally. You learned how you can use Redux to create a
-global state that handles every state aspect of your application. You now have
-the task to create these building blocks:
+the individual components locally. We will now prepare for a global state
+handling solution powered by *Redux*.
 
-* actions
-* action creators
-* reducer(s) (if you want you can create more than one reducer, but this is up
-  to you)
+We'll start out with the state handling itself, *without integrating it into the
+React app yet*.
 
-Since reducers are always pure functions without side effects it is very
-efficient and easy to unit test them. Write some unit tests that make sure they
-work as intended!
+Create the actions and a reducer for everything that can happen in the app:
```

Comment:
```suggestion
Create the actions and reducers for everything that can happen in the app:
```

Reply:
I was thinking about that, and I think it should be just reducer.

In plain Redux, this thing looks like:

```typescript
export function reducer(climate: ClimateState = initialState, action: ClimateActions): ClimateState {
    switch (action.type) {
        case SET_TEMPERATURE:
            return {
                ...climate,
                temperature: _updateTuple(action.temperature, climate.temperature)
            };
        case SET_HUMIDITY:
             // …
        case RESET:
             // …
        default:
             // …
    }
}                                                                                        
```

So it's just a single reducer.

Reply:
yeah, I think it would only lead to confusion if we write "reducers"